### PR TITLE
Add implementation for __pow__

### LINKED ISSUE #36: Complete implementation of operations on standard types
Java has a specific set of allowed operator and operations. Python has a set of operators and operations as well. However, the two don't match exactly. For example, Java will allow any object to be added to a `java.lang.String`; Python does not. On the other hand, Python allows multiplication of `str` by `int` (producing a duplicated string); Java does not. 

In order to replicate Python behavior in Java, Python's logic for all the basic operations needs to be implemented in the VOC Java support library.

The test suite contains around 2800 tests in the that are currently marked as "expected failures". These reflect an attempt to do "every operation between every base data type". The task: pick a data type, and write the implementation of one of the dunder operation (e.g., `__add__` or `__xor__`) or comparison (`__lt__` or `__eq__`) methods. 

The tests will pass (and become unexpected successes) when the output of Python code doing that operation is the same when run through CPython and VOC - and I mean byte-identical, down to the text and punctuation of the error message if appropriate.

If you want to see an example of what is involved, check out the implementation of [__add__](https://github.com/pybee/voc/blob/master/python/common/org/python/types/Int.java#L135) for integers. If you have an `int`, Python will allows you to add an `int` or a float` to it; all other types raise a `TypeError`. The list of operations on `int` that still need to be implemented can be [found here](https://github.com/pybee/voc/blob/master/tests/datatypes/test_int.py); if you add a new operation, delete the line that corresponds to that test, and the test will report as a pass, rather than an expected fail.

### PR DESCRIPTION
**Pull request by ganeshhubale**
Fixes #36

### CODE REVIEW

**Comment by onyb** [comment]
@ganeshhubale The PR doesn't fix anything, as the check you introduced is already taken care of here: https://github.com/pybee/voc/blob/master/python/common/org/Python.java#L1308-L1328

Ideally, the `pow()` implementation linked above should do nothing more than calling `__pow__`. All the checks should instead be performed in each `__pow__` implementation. Please take a look at the following examples to know why. Note that this issue is not just limited to `pow()`.

The following works in Python:
```py
class HypotheticalInt:
    def __pow__(self, y, z):
        return 'Hypothetical result'

>>> pow(1, -2, 3)
TypeError: pow() 2nd argument cannot be negative when 3rd argument specified
>>> x = HypotheticalInt()
>>> pow(x, -2, 3)
'Hypothetical result'
```

But the same fails on VOC with an error, albeit a different one.

```py
>>> x = HypotheticalInt()
>>> pow(x, -2, 3)
TypeError: pow() 3rd argument not allowed unless all arguments are integers
```

File: python/common/org/python/types/Float.java [hunk 1]
```diff
@@ -450,7 +450,10 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
     )
     public org.python.Object __pow__(org.python.Object other, org.python.Object modulo) {
         if (modulo != null) {
-            throw new org.python.exceptions.NotImplementedError("float.__pow__() with modulo has not been implemented");
+            //throw new org.python.exceptions.NotImplementedError("float.__pow__() with modulo has not been implemented");
+            if (other instanceof org.python.types.Float) {
```
Comment (onyb) [inline python/common/org/python/types/Float.java]: Actually you don't need to perform this check, as `this` itself is a `Float` and using `modulo` should be disallowed regardless of whether `other` is an instance of `org.python.types.Float` or not.

**Comment by eliasdorneles** [comment]
@onyb I do agree that fixing the issue you mention makes sense -- I'm not sure what's the best approach, but you're welcome to try something. :)

**Comment by eliasdorneles** [comment]
@onyb you might want to have a look into the proof-of-concept in #651

**Pull request closed without merging**